hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ScmTableCountTask.java

```diff
@@ -0,0 +1,217 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.recon.tasks;
+
+import com.google.common.collect.Iterators;
+import org.apache.hadoop.hdds.utils.db.DBStore;
+import org.apache.hadoop.hdds.utils.db.Table;
+import org.apache.hadoop.hdds.utils.db.TableIterator;
+import org.apache.hadoop.ozone.recon.scm.ReconScmTask;
+import org.apache.hadoop.ozone.recon.scm.ReconStorageContainerManagerFacade;
+import org.hadoop.ozone.recon.schema.UtilizationSchemaDefinition;
+import org.hadoop.ozone.recon.schema.tables.daos.ReconTaskStatusDao;
+import org.hadoop.ozone.recon.schema.tables.daos.ScmTableCountDao;
+import org.hadoop.ozone.recon.schema.tables.pojos.ScmTableCount;
+import org.jooq.DSLContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+import java.util.HashMap;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Iterator;
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import static org.apache.hadoop.hdds.scm.metadata.SCMDBDefinition.DELETED_BLOCKS;
+import static org.hadoop.ozone.recon.schema.UtilizationSchemaDefinition.SCM_TABLE_COUNT_TABLE_NAME;
+
+
+/**
+ * Any background task that tracks SCM's table counts.
+ */
+public class ScmTableCountTask extends ReconScmTask {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(ScmTableCountTask.class);
+
+  private final DBStore scmDBStore;
+  private final ScmTableCountDao scmTableCountDao;
+  private final DSLContext dslContext;
+  private final long interval;
+  private ReadWriteLock lock = new ReentrantReadWriteLock(true);
+  private static final long NANOSECONDS_IN_MILLISECOND = 1_000_000;
+
+  public ScmTableCountTask(ReconStorageContainerManagerFacade reconSCM,
+                           ReconTaskStatusDao reconTaskStatusDao,
+                           ReconTaskConfig reconTaskConfig,
+                           ScmTableCountDao scmTableCountDao,
+                           UtilizationSchemaDefinition schemaDefinition) {
+    super(reconTaskStatusDao);
+    this.scmDBStore = reconSCM.getScmDBStore();
+    this.scmTableCountDao = scmTableCountDao;
+    this.dslContext = schemaDefinition.getDSLContext();
+    this.interval = reconTaskConfig.getScmTableCountTaskInterval().toMillis();
+  }
+
+  @Override
+  protected synchronized void run() {
+    try {
+      while (canRun()) {
+        wait(interval);
+        runTableCountProcess();
+      }
+    } catch (Throwable t) {
+      LOG.error("Error while running ScmTableCountTask: {}", t);
+      if (t instanceof InterruptedException) {
+        Thread.currentThread().interrupt();
+      }
+    }
+  }
+
+  private void runTableCountProcess() {
+    try {
+      long startTime, endTime, duration, durationMilliseconds;
+      try {
```

Comment:
Yea it makes sense! Since the inner try block is only catching an Exception and the outer try block is catching Throwable, there is no specific need for them to be separate.